public/js/login.js

```diff
@@ -0,0 +1,32 @@
+const loginFormHandler = async (event) => {
+  event.preventDefault();
+
+  // trim = leerzeichen vorne und hinten abschneiden
+  const username = document.querySelector("#username").value.trim();
+  const password = document.querySelector("#password").value.trim();
+
+  if (username && password) {
+    const response = await fetch("/api/user/login", {
+      method: "POST",
+      body: JSON.stringify({ username, password }),
+      headers: { "Content-Type": "application/json" },
+    });
+
+    if (response.ok) {
+      // Wenn alles okay, dann geht er direkt zum Dashboard
+      document.location.replace("/dashboard");
+      const dashboardNode = document.querySelector("#dashboard");
+      console.log(dashboardNode);
+      dashboardNode.classList.remove("disabled");
+      const logoutNode = document.querySelector("#logout");
+      logoutNode.classList.remove("disabled");
+    } else {
+      alert("Failed to log in.");
```

Comment:
From a UX standpoint, it may be a little better for the user to see an error message in place of an alert box if the login is not successful. Another nice addition could be specifying why it was not successfull e.g. is the password incorrect or does the user not exist?